src/contracts/Output/VisitorAdapterNormalizer.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Contracts\Rest\Output;
+
+use LogicException;
+use Symfony\Component\Serializer\Normalizer\ContextAwareNormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareTrait;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+final class VisitorAdapterNormalizer implements NormalizerInterface, NormalizerAwareInterface, ContextAwareNormalizerInterface
+{
+    use NormalizerAwareTrait;
+
+    private const string CALLED_CONTEXT = __CLASS__ . '_CALLED';
```

Comment:
@alongosz we might want to standardize naming for encoder key, when it should be skipped in the next normalization pass, just like in your PR.